packages/snjs/lib/services/singleton_manager.ts

```diff
@@ -173,14 +173,7 @@ export class SNSingletonManager extends PureService {
      * of a download-first request.
      */
     if (handled.length > 0 && eventSource === SyncEvent.FullSyncCompleted) {
-      /**
-       * Do not await. We want any local-side changes to
-       * be awaited but the actual sync shouldn't be since it's non-essential
-       * Perform after timeout so that we can yield to event notifier that triggered us
-       */
-      setTimeout(() => {
-        this.syncService?.sync();
-      });
+      await this.syncService?.sync();
```

Comment:
I assume the situation has changed?

Reply:
It was difficult to create tests where you can be sure that there is no background sync happening without awaiting here. And I think in general all sync calls should be awaitable and not detached. And no test failed by awaiting here 🤷‍♀️